### PR TITLE
CNF-8518: ztp: Update DefaultCatSrc with Status and add depreciation warning in the rest of the cat sources

### PR DESCRIPTION
**Pull request by pixelsoccupied**
Update DefaultCatSrc with Status and add depreciation warning in rest of the CatalogueSource (we are meant to use just the  one)
/cc @imiller0 
/cc @jc-rh

### CODE REVIEW

**Review by imiller0 — COMMENTED** [review]
/lgtm
Perhaps we should go ahead and deprecate all but DefaultCatSource now? It is confusing to have only 1 catalogsource in the reference configuration but carry one per subscription.

**Comment by pixelsoccupied** [comment]
> deprecate all but DefaultCatSource

@imiller0 I removed all the CRs that are not in reference. Not sure if there's any way to add a depreciation warning with the current stack before removing though. wdyt?

File: ztp/policygenerator/utils/utils.go [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
+	DefaultCompliantEvaluationInterval    = "10m"
+	DefaultNonCompliantEvaluationInterval = "10s"
+	DisableEvaluationInterval             = "never"
+	// DefaultComplianceType ComplianceType of "mustonlyhave" uses significant CPU to enforce. Default to
+	// "musthave" so that we realize the CPU reductions unless explicitly told otherwise
+	DefaultComplianceType = "musthave"
+	// ZtpDeployDeprecationWarningKey annotation key for when source-cr is set to be deprecated. See ZtpDeployDeprecatedSourceCRFileNames for full list
+	ZtpDeployDeprecationWarningKey = "ran.openshift.io/deprecated"
+	// ZtpDeployDeprecationWarningValue annotation value for when source-cr is set to be deprecated. See ZtpDeployDeprecatedSourceCRFileNames for full list
+	ZtpDeployDeprecationWarningValue = "true"
```
Comment (imiller0) [inline ztp/policygenerator/utils/utils.go:22]: Would it make sense to have a longer message here to indicate why/when it is deprecated. ie, some sort of informational text:
`ran.openshift.io/deprecation-warning: "Source CR StorageCatSource is deprecated. Use DefaultCatSource instead"`

**Review by imiller0 — COMMENTED** [review]
I think we need to consider what happens when managing multiple versions simultaneously. policyGen may be building policies from 4.n source CRs and 4.n+1 source CRs simultaneously. In that case the n+1 should contain the deprecation warning and the n should not. Maybe we can move the annotation into the source CR and trigger off of that?

**Review by imiller0 — APPROVED** [review]
/lgtm

**Pull request merged**